features/steps/utils/misc.py

```diff
@@ -24,7 +24,8 @@ def inner(*args):
 
 def do_try(fn, default=None):
     try:
-        return fn()
+        result = fn()
+        return result if result else default
```

Comment:
The idea was because of this line
https://github.com/buildingSMART/ifc-gherkin-rules/blob/53c803f2838ee311d1e949d4af61a796d3f1525a/features/steps/thens/attributes.py#L24-L26

I was rewriting the step implementation to iterate over every `instance` explicitly.
In the altered implementation, the function would executes and returns `None` instead of `'Not found`', as was the case before the rewrite. Pragmatically, it does not matter, because in both cases `entity_obj_placement` is not equal to `entity_obj_placement_rel`. However, I found it weird to leave a 'Not found' that is not used anymore, and I wanted to change as little to the code as possible (especially since I didn't write this rule myself). But the `do_try` func is used quite often in different scenario's, so let's indeed leave that unchanged.

Looking at it now, I see that we can prevent altering `do_try` and simplify it to
`entity_obj_placement_rel = getattr(related_obj_placement, "PlacementRelTo", None)`

Changed it in [reverse do_try changes and simplify step_impl of ojp001](https://github.com/buildingSMART/ifc-gherkin-rules/pull/119/commits/2add4bc9f80cfc7ef26b2600afc8ccf24cc06624)

Reply:
Yes I think do_try should really be to catch exceptions in a onelines, not deal with missing, but otherwise valid values. If you want to do that in a oneliner you could have done:

`entity_obj_placement_rel = misc.do_try(lambda: related_obj_placement.PlacementRelTo) or 'Not found' `

Leave the default=None to None, but with the `or 'Not found` suffix you get any falsy value replaced with Not found, whether it comes from the do_try exception catching or the actual value being None.